### PR TITLE
Feature/get all

### PR DESCRIPTION
**Pull request by KadirMarangoz**
## Proposed changes

Added support for a `getAllUsers` through the Auth0 Management API to both real and fake User facade including tests.

## Types of changes

What types of changes does your code introduce to this repository?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Unit testing

Run vendor/bin/phpunit from the project root

### Manual testing

Not Applicable

### CODE REVIEW

File: tests/Feature/UserRepositoryTest.php [hunk 1]
```diff
@@ -746,4 +746,47 @@ public function testDeleteOne()
         // Verify correct endpoint was called
         self::assertEquals('/api/v2/users/' . $userID, $request->getUri()->getPath());
     }
+
+    public function testGetAllUsers()
```
Comment (marijnvanderhorst) [inline tests/Feature/UserRepositoryTest.php:750]: Please add a PhpDoc comment

File: tests/Feature/UserRepositoryTest.php [hunk 2]
```diff
@@ -746,4 +746,47 @@ public function testDeleteOne()
...
+        "email":"other@gmail.com","email_verified":false,"username":"other","phone_number":
+        "+199999999999999","phone_verified":false,"created_at":"","updated_at":"","identities":[{"connection":
+        "Initial-Connection","user_id":"507f1f77bcf86cd799439020","provider":"auth0","isSocial":false}],
+        "app_metadata":{},"user_metadata":{},"picture":"","name":"","nickname":"","multifactor":[""],"last_ip":
+        "","last_login":"","logins_count":0,"blocked":false,"given_name":"","family_name":""}]')];
+
+        // Call function under test.
+        $users = $this->repo->getAllUsers();
+
+        // Expect one user returned.
```
Comment (marijnvanderhorst) [inline tests/Feature/UserRepositoryTest.php:767]:
```suggestion
        // Expect two users returned.
```

**Review by marijnvanderhorst — CHANGES_REQUESTED** [review]
Looks good, some small comments

File: tests/Feature/UserRepositoryTest.php (hunk 1 above)
Comment (KadirMarangoz) [reply tests/Feature/UserRepositoryTest.php:750]: Fixed

File: src/Contracts/UserRepository.php [hunk 3]
```diff
@@ -25,6 +25,15 @@ public function get($id): ?object;
      */
     public function getByIds(Collection $ids, array $fields = null): Collection;
 
+    /**
+     * Retrieve all users within Auth0, optionally limited to only contain the given fields.
+     * The returned collection is keyed by the email addresses.
+     *
+     * @param array|null $fields Fields to be retrieved for each user.
```
[the remark this answers is not in the document]
Comment (marijnvanderhorst) [reply src/Contracts/UserRepository.php]: Comment above it is still incorrect: "optionally limited..."

**Review by marijnvanderhorst — APPROVED** [review]
Looks good 👍

**Pull request merged**